packages/model-viewer/src/three-components/SmoothControls.ts

```diff
@@ -839,33 +864,168 @@ export class SmoothControls extends EventDispatcher {
     const {isUserChange} = this;
     this.isUserChange = true;
 
+    /* Simple if() logic to catch modifier keys
+    * There is undoubtedly a nicer way of doing this
+    * Only considering modifiers of shift+ctrl, shift, ctrl and none.
+    * This maps to pan+small, pan, (orbit)small and (orbit)
+    * Pan key choice inspired by Blender
+    */
+    if (event.shiftKey && event.ctrlKey) {
+      (this.scene.element as any)[$panElement].style.opacity = 1;
+      relevantKey = this.panKeyCodeHandler(event, PAN_INCREMENT_SMALL, PAN_INCREMENT_SMALL);
+    } else if (event.shiftKey) {
+      (this.scene.element as any)[$panElement].style.opacity = 1;
+      relevantKey = this.panKeyCodeHandler(event, PAN_INCREMENT, PAN_INCREMENT);
+    } else if (event.ctrlKey) {
+      relevantKey = this.orbitZoomKeyCodeHandler(event, ZOOM_SENSITIVITY, KEYBOARD_ORBIT_INCREMENT_SMALL, isUserChange);
+    } else {
+      relevantKey = this.orbitZoomKeyCodeHandler(event, ZOOM_SENSITIVITY, KEYBOARD_ORBIT_INCREMENT, isUserChange);
+    }
+     
+    // switch (event.keyCode) {
+    //   case KeyCode.PAGE_UP:
+    //     this.userAdjustOrbit(0, 0, ZOOM_SENSITIVITY);
+    //     break;
+    //   case KeyCode.PAGE_DOWN:
+    //     this.userAdjustOrbit(0, 0, -1 * ZOOM_SENSITIVITY);
+    //     break;
+    //   case KeyCode.UP:
+    //     this.userAdjustOrbit(0, -KEYBOARD_ORBIT_INCREMENT, 0);
+    //     break;
+    //   case KeyCode.DOWN:
+    //     this.userAdjustOrbit(0, KEYBOARD_ORBIT_INCREMENT, 0);
+    //     break;
+    //   case KeyCode.LEFT:
+    //     this.userAdjustOrbit(-KEYBOARD_ORBIT_INCREMENT, 0, 0);
+    //     break;
+    //   case KeyCode.RIGHT:
+    //     this.userAdjustOrbit(KEYBOARD_ORBIT_INCREMENT, 0, 0);
+    //     break;
+    //   default:
+    //     relevantKey = false;
+    //     this.isUserChange = isUserChange;
+    //     break;
+    // }
+
+    if (relevantKey) {
+      event.preventDefault();
+    }
+  };
+
+  private onKeyUp = (event: KeyboardEvent) => {
+    // console.log('In onKeyUp', event);
+    let opacity = 1;
+    // Only hide $panElement if shift key released (#16) or .shiftKey is false (although not sure if this second case ever gets hit when opacity == 1)
+    if (event.keyCode == KeyCode.SHIFT){
+      opacity = 0;
+      // console.log('Released shift');
+    } else if (!event.shiftKey) {
+      opacity = 0;
+      // console.log('Released something else');
+    }
+    (this.scene.element as any)[$panElement].style.opacity = opacity;
+  }
+
+  private orbitZoomKeyCodeHandler(event: KeyboardEvent, ZoomInc: number, OrbitInc: number, isUserChange: boolean) {
```

Comment:
Thanks; would you mind writing out just the behaviors you desire for each user input? Then I at least know the UX you're going for and we can figure out the easiest way to accomplish it.

Reply:
In words...
As the user I want to be able to use the keyboard to orbit, pan and zoom.
Because: I can already do all these things with touch and mouse, but only pan with keys and for some, but not all models, being able to step a change, rather than drag a change, would be the preferred way of nudging or changing the view.

As a developer I want to be able to offer the keyboard interaction above and in addition offer two "steps sizes" as watching users interacting with the initial stepping controls reveals that there is no right size step and a single size is either too coarse or too fine. 

At the same time the idea of coarse/fine step sizes nicely applies to the mouse controls (in the sense that it's not weird and once seen gets an oh yeah, got it response).

Reply:
In words...
As the user I want to be able to use the keyboard to orbit, pan and zoom.
Because: I can already do all these things with touch and mouse, but only pan with keys and for some, but not all models, being able to step a change, rather than drag a change, would be the preferred way of nudging or changing the view.

As a developer I want to be able to offer the keyboard interaction above and in addition offer two "steps sizes" as watching users interacting with the initial stepping controls reveals that there is no right size step and a single size is either too coarse or too fine.

At the same time the idea of coarse/fine step sizes nicely applies to the mouse controls (in the sense that it's not weird and once seen gets an oh yeah, got it response).